app/javascript/javascript/sessionsMap.js

```diff
@@ -0,0 +1,339 @@
+import * as FiltersUtils from "./filtersUtils";
+import { clearMap } from "./clearMap";
+import { applyTheme } from "./theme";
+import { getParams } from "./params";
+import sensors from "./sensors";
+import pubsub from "./pubsub";
+import params from "./params2";
+import map from "./map";
+import updateCrowdMapLayer from "./updateCrowdMapLayer";
+import constants from "./constants";
+const sessions =
+  window.location.pathname === constants.mobileMapRoute
+    ? require("./mobileSessions").default
+    : require("./fixedSessions").default;
+
+export default (() => {
+  let pulsatingSessionMarker = null;
+  const elmApp = window.__elmApp;
+  sensors.setSensors(window.__sensors);
+
+  function setDefaults() {
+    clearMap();
+    map.unregisterAll();
+
+    const sensorId = params.get("data", { sensorId: sensors.defaultSensorId })
+      .sensorId;
+
+    const defaults = {
+      sensorId,
+      location: "",
+      tags: "",
+      usernames: "",
+      timeFrom: FiltersUtils.oneYearAgo(),
+      timeTo: FiltersUtils.endOfToday(),
+      heat: {
+        lowest: 0,
+        low: 12,
+        mid: 35,
+        high: 55,
+        highest: 150
+      }
+    };
+
+    const defs = sessions.isMobile()
+      ? { ...defaults, gridResolution: 31, crowdMap: false }
+      : { ...defaults, isIndoor: false, isActive: true };
+
+    params.updateFromDefaults(defs);
+  }
+
+  setDefaults();
+
+  pubsub.subscribe("googleMapsChanged", function(newValue) {
+    if (params.isSessionSelected()) return;
+    // when loading the page for the first time sometimes the watch is triggered twice, first time with hasChangedProgrammatically as undefined
```

Comment:
is this still the case?

Reply:
I don't think so but I didn't want to force my hand too much in this PR ;)

Reply:
In any case, good catch!